Tensile/Source/client/source/Reference.cpp

```diff
@@ -118,6 +118,60 @@ namespace Tensile
             throw std::runtime_error(msg.c_str());
         }
 
+        /* high precision accumulate: dist type != Accumulator or src type != Accumulator, such as H/H/H + HPA, H/H/S + HPA, B/B/S + HPA, B/S/S + HPA
+         * Ignore types: I8/I32/I32, I8x4/I32/I32
+         */
+        template <typename Inputs, typename Accumulator>
+        constexpr bool is_hpa
```

Comment:
Why not just check is_hpa = (D Type !=AccType)?

Reply:
Consider the case, B/S/S +HPA,
both types of D and ACC are S. If we just check the D and ACC types then the _is_hpa()_ will return false.
However, in this example, the HPA is true and we want to raise the precision when doing _rC = A x B. => rC = float(A) x float(B)_.
So, extended to check the type of A and B will be helpful.